File: article/static/article/columns.js
/**
 * Created by Matthias on 11/10/2016.
 */

import { Column } from 'js/tools/tables'



class ProductLabelsColumn extends Column {
  CellRenderer() {
    let self = this;

    /**
     * @returns {*[]}
     */
    function render(labels) {
      return [
        'td', {
        },
        labels.map((label) => [[
          'span', {},
          label
        ], ' '])
      ];
    }
    return {
      render: render
    };
  }
}

File: article/columns.py
from django.template import Template, Context
from django.utils.translation import ugettext_lazy as _

from tools.tables import Column


class ProductLabelsColumn(Column):
    """
        Product label column. Point it the labelset of a product, and it will render all labels that are known.
    """
    cell_template = Template("""{% load i18n %}
                                <td>{% for label in labels %}
                                    <span class="label">
                                        {{ label }}
                                    </span> {% empty %}
                                    {% trans 'No labels' %}
                                {% endfor %}</td>""")
    name = _('Labels')

    def get_context(self, row):
        return Context({'labels': row})
